sdk/tests/core/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ if (UNIX)
     set(MATH_LIB_UNIX "m")
 endif()
 
-if (AZ_PLATFORM_IMPL STREQUAL "POSIX" AND NOT APPLE AND UNIT_TESTING_MOCKS)
+if (AZ_PLATFORM_IMPL STREQUAL "POSIX" AND NOT APPLE AND UNIT_TESTING_MOCKS AND NOT AZ_MQTT_TRANSPORT_IMPL STREQUAL "PAHO")
```

Comment:
Why are these tests disabled when PAHO MQTT is used?

Reply:
MQTT testing is disabled for rpc tests on Paho. We should not be using any underlying MQTT implementation for the tests in core but leaving as a TODO for later

Reply:
I'm confused - then why is it disabling the tests if AZ_MQTT_TRANSPORT_IMPL is set to "PAHO" but not "MOSQUITTO"?